src/layouts/Sponsor.tsx

```diff
@@ -73,13 +79,49 @@ export function Sidebar({
     onClose: onSponsorInfoModalClose,
   } = useDisclosure();
 
-  useEffect(() => {
-    if (
-      userInfo?.currentSponsorId &&
-      (!userInfo?.firstName || !userInfo?.lastName || !userInfo?.username)
-    ) {
-      onSponsorInfoModalOpen();
+  const {
+    isOpen: isScoutAnnounceModalOpen,
+    onOpen: onScoutAnnounceModalOpen,
+    onClose: onScoutAnnounceModalClose,
+  } = useDisclosure();
+
+  function sponsotInfoCloseAltered() {
+    onSponsorInfoModalClose();
+    onScoutAnnounceModalOpen();
+  }
+
+  const getSponsorLatestActiveSlug = async () => {
+    try {
+      const slug = await axios.get('/api/bounties/latestActiveSlug');
+      if (slug.data) {
+        setLatestActiveSlug(slug.data.slug);
+      }
+    } catch (e) {
+      console.log(e);
     }
+  };
+
+  useEffect(() => {
+    const modalsToShow = async () => {
+      if (
+        userInfo?.currentSponsorId &&
+        (!userInfo?.firstName || !userInfo?.lastName || !userInfo?.username)
+      ) {
+        onScoutAnnounceModalClose();
```

Comment:
why close the modal? opening it only if all conditions are met is a better idea right?

Reply:
yea same thing, must've missed to remove this line

Reply:
u are right, pushed to remove this